### PR TITLE
fix: Add tests for setNetworkClientIdForDomain being called on all network items

### PR DESCRIPTION
**Pull request by darkwing**
## **Description**

Adds tests for the functionality added in https://github.com/MetaMask/metamask-extension/pull/25260

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25270?quickstart=1)

## **Related issues**

Fixes: N/A

## **Manual testing steps**

1. Run jest tests

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by salimtb — APPROVED** [review]

**Comment by salimtb** [comment]
@darkwing test cherry-picked to this [PR](https://github.com/MetaMask/metamask-extension/pull/25259)

File: ui/components/multichain/network-list-menu/network-list-menu.test.js [hunk 1]
```diff
@@ -150,4 +158,22 @@ describe('NetworkListMenu', () => {
...
+      NETWORK_TYPES.MAINNET,
+    );
+  });
+
+  it('fires setNetworkClientIdForDomain when test network item is clicked', () => {
+    const { getByText } = render({ showTestNetworks: true });
+    fireEvent.click(getByText(SEPOLIA_DISPLAY_NAME));
+    expect(mockSetNetworkClientIdForDomain).toHaveBeenCalledWith(
+      MOCK_ORIGIN,
+      NETWORK_TYPES.SEPOLIA,
```
Comment (salimtb) [inline ui/components/multichain/network-list-menu/network-list-menu.test.js:176]: i guess we should add this code on testnet ( see it [here](https://github.com/MetaMask/metamask-extension/pull/25259/commits/bc8a7fa6a236423ebccc48f0010da067c13c4c34#diff-1832ff00c884a64a41fb0eabeb10d3472192e1f5dd4a3e36018f5d4706944bd0R312-R317) )

**Review by EtherWizard33 — APPROVED** [review]
LGTM

File: ui/components/multichain/network-list-menu/network-list-menu.test.js (hunk 1 above)
Comment (salimtb) [reply ui/components/multichain/network-list-menu/network-list-menu.test.js:176]: for more context see the bug below:

https://github.com/MetaMask/metamask-extension/assets/26223211/a940295c-c71f-4aff-8c40-65769978055c

**Comment by darkwing** [comment]
Closing as test was used in https://github.com/MetaMask/metamask-extension/pull/25646

**Pull request closed without merging**